jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java

```diff
@@ -1342,8 +1378,46 @@ protected HttpClient getHttpClient(Integer readTimeout,
             return httpClient;
         }
     }
-    
-    private HttpRoutePlanner getRoutePlanner ()
+
+	protected BasicCookieStore addCookie(String url, String cookie, String cookiePath) {
+		String host = null;
+		BasicClientCookie basicClientCookie = null;
+		BasicCookieStore basicCookieStore = null;
+
+		try {
+			host = new URL(url).getHost();
+		} catch (MalformedURLException e) {
+			host = null;
+			logger.error("Unexpected exception occured while retriving host", e);
+			return null;
+		}
+		basicCookieStore = new BasicCookieStore();
+		for (String c : cookie.split(";")) {
+			String[] cookieParts = c.split("=", 2);
+			if (cookieParts.length == 2 && validateCookie(cookieParts)) {
+				basicClientCookie = new BasicClientCookie(cookieParts[0], cookieParts[1]);
+				basicClientCookie.setDomain(host);
+				if (cookiePath != null && !cookiePath.isEmpty()) {
+					basicClientCookie.setPath(cookiePath);
+				} else {
+					basicClientCookie.setPath("/");
+				}
+				basicCookieStore.addCookie(basicClientCookie);
+			}
+		}
+		return basicCookieStore;
+	}
+
+	private boolean validateCookie(String[] cookieParts) {
+		if ((!cookieParts[0].trim().isEmpty() && !cookieParts[0].equals("null"))
+				&& (!cookieParts[1].trim().isEmpty() && !cookieParts[1].equals("null"))) {
```

Comment:
The comparison with "null" seems to be set ad-hoc for the tests. This workitemhandler receives the cookies in a single string to be splitted (with ";" as separator): if the client sets "key1=null" as cookie, it should be sent that way,

Reply:
Modified code to accept key1=null cookie